### DISCUSSION TITLE
"worker call timed out!" Error and freezing when querying one of my endpoints

### DISCUSSION THREAD

**Discussion by enpro-github** [Budibase/budibase#4213] [Help] [answered]
I have a REST datasource set up that queries an API I have created. One of the datasources I'm using for a POST request to the same API works perfectly. However since updating to the newest version of budibase my GET request will cause Budibase to freeze when I use `return data` in the transformer field. And if I try `return typeof data` (was just playing around with different solutions) I get `worker call timed out!`. This was all working fine a few days ago before updating. If I try to reference a field inside I'm getting `undefined` when that never happened before. 

Thank you!

**Comment by mike12345567** [comment]
Hi @enpro-github - just to check, is this self hosted or in the Budibase Cloud?

Is it possible your GET API request is slow to respond? By default we've added a timeout to queries, however in self host you can expand the timeout by setting the `QUERY_THREAD_TIMEOUT` environment variable, for example `QUERY_THREAD_TIMEOUT=30000` would give the queries 30 seconds to complete.

**Comment by enpro-github** [comment]
The API never had an issue with Budibase before, also I can query it through Google Chrome no problem. Yes I am self-hosting, changing the environment variable doesn’t seem to have any effect, it shows the same error after about 6 seconds. If I change my api to just always return {“test”: “test”} for example it works fine. So it seems it’s something to do with the size?

**Reply by mike12345567** [reply]
Hi @enpro-github - just to check, how did you set the environment variable? You'll need to add it to your docker environment, so in the `docker-compose.yaml` file from which you compose your setup you'll need to add the variable to the `environment` section, then restart the stack.

It'll need to look something like this:
```services:
  app-service:
    restart: always
    image: budibase.docker.scarf.sh/budibase/apps
    container_name: bbapps
    ports:
      - "${APP_PORT}:4002"
    environment:
      SELF_HOSTED: 1
      QUERY_THREAD_TIMEOUT: 30000
      ......
```

If it is still timing out after 6 seconds I expect this is the issue, it hasn't actually started using the new query timeout length.

**Comment by enpro-github** [comment, marked as answer]
Oh I actually set it by the normal method with Linux. Thank you for the update. I actually figured out the issue, I needed to convert the data returned to a dictionary. However I did not need to do this before, you guys didn’t change anything did you?

**Reply by mike12345567** [reply]
Hi @enpro-github - glad you found the issue.

We re-wrote our REST API datasource as part of the latest release, previously it was a little light on error checking and may have missed some things that could have caused edge-case errors down the line. Unsure what was causing it to timeout however, we now have the ability to process many more body types than before - we'll need to look at adding better error logging out of the queries so that users can see what specific aspect of the request is causing an issue for Budibase.

**Comment by enpro-github** [comment]
Thank you! And yes I'd definitely agree with that, the problem was that the transformer was trying to parse a string like it was a dict. Took me days to fix lol, kinda stupid on my part. But I'd definitely say that would be a simple error to display

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
